lib/hashgraph.cc

```diff
@@ -276,29 +276,22 @@ void Hashgraph::consume_sequence_and_tag(const std::string& seq,
 //
 
 // TODO? Inline in header.
-void
-Hashgraph::
-consume_fasta_and_tag(
-    std:: string const  &filename,
-    unsigned int	      &total_reads, unsigned long long	&n_consumed
+template<typename SeqIO>
+void Hashgraph::consume_fasta_and_tag(
+        std::string const &filename,
+        unsigned int &total_reads,
+        unsigned long long &n_consumed
 )
 {
-    IParser *	  parser =
-        IParser::get_parser( filename );
-
-    consume_fasta_and_tag(
-        parser,
-        total_reads, n_consumed
-    );
-
-    delete parser;
+    ReadParserPtr<SeqIO> parser = get_parser<SeqIO>((std::string&)filename);
```

Comment:
I'm using casting to eliminate errors with inconsistent use of `const` in the function call tree. I'll see if I can revisit this and make it cleaner.